bindings/node/lib/clientEncryption.js

```diff
@@ -549,6 +558,65 @@ module.exports = function (modules) {
       return value;
     }
 
+    /**
+     * A convenience method for creating an encrypted collection.
+     * This method will create data keys for any encryptedFields that do not have a `keyId` defined
+     * and then create a new collection with the full set of encryptedFields.
+     *
+     * @template {TSchema} - Schema for the collection being created
+     * @param {Db} db - A database to create the collection in
+     * @param {string} name - The name of the collection to be created
+     * @param {object} options - Options for createDataKey and for createCollection.
+     * @param {string} options.provider - KMS provider name
+     * @param {AWSEncryptionKeyOptions | AzureEncryptionKeyOptions | GCPEncryptionKeyOptions} [options.masterKey] - masterKey to pass to createDataKey
+     * @param {CreateCollectionOptions} options.createCollectionOptions - options to pass to createCollection, must include `encryptedFields`
+     * @returns {Promise<{ collection: Collection<TSchema>, encryptedFields: Document }>} - created collection and generated encryptedFields
+     * @throws {MongoCryptCreateDataKeyError} - If part way through the process a createDataKey invocation fails, an error will be rejected that has the partial `encryptedFields` that were created.
+     * @throws {MongoCryptCreateEncryptedCollectionError} - If creating the collection fails, an error will be rejected that has the entire `encryptedFields` that were created.
+     */
+    async createEncryptedCollection(db, name, options) {
+      const {
+        provider,
+        masterKey,
+        createCollectionOptions: {
+          encryptedFields: { ...encryptedFields },
+          ...createCollectionOptions
+        }
+      } = options;
+
+      if (Array.isArray(encryptedFields.fields)) {
+        const createDataKeyPromises = encryptedFields.fields.map(async field =>
+          typeof field !== 'object' || field.keyId != null
+            ? field
+            : {
+                ...field,
+                keyId: await this.createDataKey(provider, { masterKey })
+              }
```

Comment:
```suggestion
  if (typeof field !== 'object' || field.keyId != null) {
    return field;
  }
  return {
                ...field,
                keyId: await this.createDataKey(provider, { masterKey })
              }
```

excuse the formatting here.  ternaries where we're returning objects and cloning are far less readable than an if-statement + return.